source/Detector/Amptek/AmptekSDD123Server.cpp

```diff
@@ -0,0 +1,516 @@
+	packetIDCounter_++;
+
+	if (sendFakeRequest) {
+		fakeSendDatagram(requestStatusPacket, overrideTimeout);
+	} else if (sendSyncRequest) {
+		requestStatusPacket.appendRelatedSyncRequestID(timeOutIDCounter_);
+		sendSyncDatagram(requestStatusPacket, overrideTimeout);
+	} else {
+		sendRequestDatagram(requestStatusPacket);
+	}
+}
+
+void AmptekSDD123Server::disableMCAMCS()
+{
+	requestDataPacket(AmptekCommandManagerSGM::RequestDisableMCAMCS);
+}
+
+
+void AmptekSDD123Server::mcaChannelsCount()
+{
+	textConfigurationReadback("MCAC");
+}
+
+void AmptekSDD123Server::presetTime()
+{
+	requestDataPacket(AmptekCommandManagerSGM::ReqeustStatusPacket);
+}
+
+void AmptekSDD123Server::requestAllTextConfigurationParameters()
+{
+	QStringList manyCommands;
+	manyCommands << "ALLP" << "CLCK" << "GAIA" << "GAIF" << "GAIN" << "HVSE" << "MCAC" << "MCAE" << "MCAS" << "PAPS" << "PDMD" << "PRER" << "PURE"
+			<< "SCAI=1" << "SCAH" << "SCAL"
+			<< "SCAI=2" << "SCAH" << "SCAL"
+			<< "SCAI=3" << "SCAH" << "SCAL"
+			<< "SCAI=4" << "SCAH" << "SCAL"
+			<< "SCAI=5" << "SCAH" << "SCAL"
+			<< "SCAI=6" << "SCAH" << "SCAL"
+			<< "SCAI=7" << "SCAH" << "SCAL"
+			<< "SCAI=8" << "SCAH" << "SCAL"
+			<< "TECS" << "THFA" << "THSL" << "TPEA" << "TPFA";
+	textConfigurationReadback(manyCommands);
+}
+
+void AmptekSDD123Server::textConfigurationReadback(QString singleCommandRequest)
+{
+	QString dataString = singleCommandRequest+"=?;";
+	requestDataPacket(AmptekCommandManagerSGM::RequestTextConfigurationReadback, dataString.toAscii().toHex());
+}
+
+void AmptekSDD123Server::textConfigurationReadback(QStringList multiCommandRequest)
+{
+	QString singleCommand;
+	QString fullCommandString;
+	foreach(singleCommand, multiCommandRequest)
+		fullCommandString.append(singleCommand+";");
+	requestDataPacket(AmptekCommandManagerSGM::RequestTextConfigurationReadback, fullCommandString.toAscii().toHex());
+}
+
+void AmptekSDD123Server::textConfigurationSet(QString singleCommandRequest)
+{
+	requestDataPacket(AmptekCommandManagerSGM::RequestTextConfiguration, singleCommandRequest.toAscii().toHex());
+}
+
+bool AmptekSDD123Server::readReplyDatagram(int &id, QByteArray &datagram)
+{
+	if(hasReplyReady_){
+		id = currentRequestPacket_.packetID();
+		datagram = currentRequestPacket_.datagram();
+		hasReplyReady_ = false;
+		return true;
+	}
+	return false;
+}
+
+void AmptekSDD123Server::setRequestIntervalTimer(int mSec)
+{
+	requestIntervalTimer_ = mSec;
+}
+
+void AmptekSDD123Server::setSpectrumPacketReceiver(QObject *spectrumPacketReceiver)
+{
+	spectrumPacketReceiver_ = spectrumPacketReceiver;
+}
+
+void AmptekSDD123Server::setConfirmationPacketReceiver(QObject *confirmationPacketReceiver)
+{
+	confirmationPacketReceiver_ = confirmationPacketReceiver;
+}
+
+void AmptekSDD123Server::sendRequestDatagram(AmptekSDD123Packet packet, int overrideTimeout)
+{
+	if(socketLocallyBusy_){
+		qDebug() << "\nPACKET QUEUE IS BUSY (send request datagram)\n";
```

Comment:
Seeing as there's an error mon already being used here, would it be proper to switch this (and the other packet queue is busy messages) to be error mon info calls?